apps/labs/posts/convoluted-story-behind-topk.md

```diff
@@ -0,0 +1,213 @@
+
+---
+
+Hi~ I'm [Jules Poon](https://juliapoo.github.io/), a Year 2 undergraduate at the National University of Singapore. This summer I had the incredible opportunity to join the Open Source internship at Quansight Labs, under the mentorship of [Ralf Gommers](https://github.com/rgommers), [Nathan Goldbaum](https://ngoldbaum.github.io/) and [Matti Picus](https://github.com/mattip) for work on NumPy, and [Evgeni Burovski](https://github.com/ev-br) for work
+on SciPy.
+
+My task for NumPy is quite generic: To contribute to NumPy. So I'm placed in a position most people just beginning in open source will find themselves in. Hopefully, this blog post can help somebody just starting with open-source to navigate projects and understand the process behind the decisions made in open-source projects.
+
+First, I want to go through some tips for contributing to NumPy.
+
+## Contributing to NumPy
+
+NumPy is a massive project, so it is easy to get lost when starting. The NumPy documentation has friendly [developer documentation](https://numpy.org/devdocs/dev/index.html) that any contributor should read, such as the [Contributor Guide](https://numpy.org/contribute/) and the [Suggested Workflow](https://numpy.org/devdocs/dev/development_workflow.html).
+
+In addition to the NumPy developer documentation and the Python API documentation, I found the [C-API Documentation](https://numpy.org/devdocs/reference/c-api/index.html) to be incredibly useful for understanding the C portion of the codebase, as well as this [NumPy Internals Page](https://numpy.org/doc/stable/dev/internals.html).
+
+As a contributor, it might be difficult to know what to work on. Aside from looking at the [issues page](https://github.com/numpy/numpy/issues/) of the repository, one can look through recent [NumPy Enhancement Proposals](https://numpy.org/neps/) and the [Current Roadmap](https://numpy.org/neps/roadmap.html) to both get an idea of the more long-term goals of the project and find things you wish to dive into.
+
+Lastly, to catch up on existing discussions or to create new ones, in addition to the repository's issues and PR pages, one can look through (and join) the [Mailing List](https://mail.python.org/mailman3/lists/numpy-discussion.python.org/). The Mailing List also announces the details of the biweekly community meetings where you can ask questions and discuss details.
+
+## My Contributions
+
+Throughout the 1.5 months working on NumPy, I made incremental improvements to the tests, documentation, CI, and submitted bug fixes. I also worked on furthering the discussion for `np.top_k` and drafted implementations, and I want to talk more about the process behind that.
+
+### Reviewing Community Discussions
+
+The discussions for `np.top_k` started from [gh-15128 Feature request: Allow np.argmax to output top K maximum values](https://github.com/numpy/numpy/issues/15128) in 2019 where multiple people expressed their request for having this feature in NumPy.
```

Comment:
Perhaps worth stating that it's the 6th-highest feature request in the whole issue tracker as measured by 👍🏼 emoji's? (that's how we looked for impactful topics)